script.reddit.reader/resources/lib/actions.py

```diff
@@ -0,0 +1,573 @@
+    li.setProperty('StartOffset', str(start_time))
+    pl.add(url, li)
+
+YTDL_VERSION_URL = 'https://yt-dl.org/latest/version'
+YTDL_LATEST_URL_TEMPLATE = 'https://yt-dl.org/latest/youtube-dl-{}.tar.gz'
+
+def ytdl_get_version_info(which_one='latest'):
+    import urllib2
+    if which_one=='latest':
+        try:
+            newVersion = urllib2.urlopen(YTDL_VERSION_URL).read().strip()
+            return newVersion
+        except:
+            return "0.0"
+    else:
+        try:
+
+            from youtube_dl.version import __version__
+            return __version__
+        except Exception as e:
+            log('error getting ytdl local version:'+str(e))
+            return "0.0"
+
+def update_youtube_dl_core(url,name,action_type):
+
+    import urllib
+    import tarfile
+
+    if action_type=='download':
+        newVersion=note_ytdl_versions()
+        LATEST_URL=YTDL_LATEST_URL_TEMPLATE.format(newVersion)
+
+        profile = xbmc.translatePath(profile_path)  #xbmc.translatePath(addon.getAddonInfo('profile')).decode('utf-8')
+        archivePath = os.path.join(profile,'youtube_dl.tar.gz')
+        extractedPath = os.path.join(profile,'youtube-dl')
+        extracted_core_path=os.path.join(extractedPath,'youtube_dl')
+
+
+        try:
+            if os.path.exists(extractedPath):
+                shutil.rmtree(extractedPath, ignore_errors=True)
+                update_dl_status('temp files removed')
+
+            update_dl_status('Downloading {0} ...'.format(newVersion))
+            log('  From: {0}'.format(LATEST_URL))
+            log('    to: {0}'.format(archivePath))
+            urllib.urlretrieve(LATEST_URL,filename=archivePath)
+
+            if os.path.exists(archivePath):
+                update_dl_status('Extracting ...')
+
+                with tarfile.open(archivePath,mode='r:gz') as tf:
+                    members = [m for m in tf.getmembers() if m.name.startswith('youtube-dl/youtube_dl')] #get just the files from the youtube_dl source directory
+                    tf.extractall(path=profile,members=members)
+            else:
+                update_dl_status('Download failed')
+        except Exception as e:
+            update_dl_status('Error:' + str(e))
+
+        update_dl_status('Updating...')
+
+        if os.path.exists(extracted_core_path):
+            log( '  extracted dir exists:'+extracted_core_path)
+
+            if os.path.exists(ytdl_core_path):
+                log( '  destination dir exists:'+ytdl_core_path)
+                shutil.rmtree(ytdl_core_path, ignore_errors=True)
+                update_dl_status('    Old ytdl core removed')
+                xbmc.sleep(1000)
+            try:
+                shutil.move(extracted_core_path, ytdl_core_path)
+                update_dl_status('    New core copied')
+                xbmc.sleep(1000)
+                update_dl_status('Update complete')
+                xbmc.sleep(2000)
```

Comment:
Just FYI it is a good practice to use the `xmbc.Monitor().waitforabort(time)` instead of `xbmc.sleep()` for times bigger than 1sec. Since kodi waits for 5 secs for the plugins to respond I don't think this is a problem. But, nevertheless, if you can adopt the waitforabort it would be great.